src/drivers/clock/sp804_timer/sp804_timer.c

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file sp804_timer.c
+ * @brief ARM Dual-Timer (SP804) Module driver
+ * @author Nishant Malpani <nish.malpani25@gmail.com>
+ * @date 2020-03-07
+ */
+
+#include <drivers/common/memory.h>
+#include <hal/clock.h>
+#include <hal/reg.h>
+#include <kernel/irq.h>
+#include <kernel/printk.h>
+#include <kernel/time/clock_source.h>
+#include <kernel/time/time_device.h>
+
+#include <framework/mod/options.h>
+#include <embox/unit.h>
+
+#define SP804_BASE      OPTION_GET(NUMBER, base_addr) /* Base address */
+#define SP804_IRQ       OPTION_GET(NUMBER, irq_nr) /* Interrupt Request number */
+#define SP804_TARGET_HZ OPTION_GET(NUMBER, target_freq) /* Frequency to run the timer at */
```

Comment:
I think you should use SYS_CLOCK instead. It's defined in hal./system.h
and setup in config as `include embox.arch.system(core_freq=48054841)`

Reply:
You mean set the target frequency of the timer same as the core's frequency? I thought all these timers are external and operate at a lower frequency than the master clock. I've used other drivers implemented as a reference [1] [2].

[1] https://github.com/embox/embox/blob/master/src/drivers/clock/cortexa9_timer/cortexa9_timer.c#L40
[2] https://github.com/embox/embox/blob/master/src/drivers/clock/epit/epit.c#L22